### ISSUE TITLE
Iteration Plan for <Month> 2022 [DRAFT]

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#1709]
This plan captures our work in **June**. This is a 4-week iteration. We will ship in early **July**.

-- Endgame
- **June 27, 2022**: Endgame begins
- **July 1, 2022**: Endgame done

The endgame details for this iteration are tracked [here](https://github-com/microsoft/vscode/-?q=is%3Aissue+label%3Aendgame-plan+milestone%3A%22June+2022%22).

-- Plan Items

Below is a summary of the top level plan items.

Legend of annotations:

| Mark                 | Description                                       |
| -------------------- | ------------------------------------------------- |
| :runner:             | work in progress                                  |
| :hand:               | blocked task                                      |
| :muscle:             | stretch goal for this iteration                   |
| :red_circle:         | missing issue reference                           |
| :large_blue_circle:  | more investigation required to remove uncertainty |
| :black_circle:       | under discussion within the team                  |
| :black_large_square: | a large work item, larger than one iteration      |

--- Accessibility
- [ ] :runner: Accessibility -, see [query](https://github-com/Microsoft/vscode/-?q=is%3Aopen+is%3Aissue+label%3Aaccessibility+milestone%3A%22June+2022%22) -isidorn **team**

--- UX
- [ ] :runner: Improve the discoverability of filtering in tree views [vscode-146806](https://github-com/microsoft/vscode/-/146806) -misolori -joaomoreno
- [ ] :muscle: Explore "Do Not Disturb" mode and its implications on existing scenarios [vscode-144324](https://github-com/microsoft/vscode/-/144324) -daviddossett -bpasero
- [ ] :red_circle: :muscle: Improve editor group border contrast -daviddossett

--- Workbench
- [ ] :runner: Support profile management [vscode-116740](https://github-com/microsoft/vscode/-/116740) -sana-ajani -sandy081 -daviddossett
- [ ] Profiles: Support global state isolation per profile [vscode-151490](https://github-com/microsoft/vscode/-/151490) -bpasero -sandy081
- [ ] :red_circle: :muscle: Support hiding toolbar items using right click -jrieken
- [ ] :runner: Explore `Continue On...` support in core [vscode-141293](https://github-com/microsoft/vscode/-/141293-issuecomment-1117468466) -joyceerhl
- [ ] :runner: Explore a `Share as...` menu for creating `vscode.dev` links [vscode-146309](https://github-com/microsoft/vscode/-/146309) -alexr00
- [ ] :runner: :muscle: Adopt inline suggestions for previewing emmet expansions [vscode-139247](https://github-com/microsoft/vscode/-/139247) -rzhao271 -hediet
- [ ] :runner: Polish Command Palette launcher in titlebar [vscode-149445](https://github-com/microsoft/vscode/-/149445) -jrieken -sbatten
- [ ] :runner: Settings Editor: Improve setting override indicators UX when many items are present [vscode-151787](https://github-com/microsoft/vscode/-/151787) -rzhao271
- [ ] :runner: Settings Editor: Indicate if setting has a language-specific value [vscode-145731](https://github-com/microsoft/vscode/-/145731)  -rzhao271
- [ ] :hand: Adopt Window Control Overlay API in electron 18 [vscode-147947](https://github-com/microsoft/vscode/pull/147947) -sbatten

--- Web
- [ ] :runner: Continue translations in the web [vscode-82595](https://github-com/microsoft/vscode/-/82595) -tylerleonhardt
  - [ ] Support built-in extension translations in web
  - [ ] Support marketplace extenson translations in web
- [ ] :runner: Explore how a web extension can leverage the [WebUSB API](https://wicg.github.io/webusb) [vscode-116761](https://github-com/microsoft/vscode/-/116761) -mjbvz -alexdima
- [ ] :red_circle: Support default themes in `/theme` -aeschli
- [ ] :runner: Explore a partial WASI implementation using the VS Code API [vscode-149092](https://github-com/microsoft/vscode/-/149092) -dbaeumer
- [ ] :red_circle: Explore setting COOP and COEP headers on vscode.dev -jrieken

--- Code Editor
- [ ] :runner: Merge editor [vscode-146091](https://github-com/microsoft/vscode/-/146091) -hediet -jrieken -bpasero
- [ ] Context menu on the minimap to hide/show it [vscode-148131](https://github-com/microsoft/vscode/-/148131) -alexdima
- [ ] Investigate providing a default snippet for a language [vscode-145929](https://github-com/microsoft/vscode/-/145929) -jrieken -digitarald
- [ ] :red_circle: Eliminate notification that shows when there are too many folding regions -aeschli

--- Comments
- [ ] Allow to disable comments altogether [vscode-149176](https://github-com/microsoft/vscode/-/149176) -alexr00
- [ ] :muscle: Comments view should have ability to filter by resolved/unresolved [vscode-150958](https://github-com/microsoft/vscode/-/150958) -alexr00

--- Notebook Editor
- [ ] Retain controller on `saveAs` or `save` of an untitled notebook [vscode-151158](https://github-com/microsoft/vscode/-/151158) -roblourens
- [ ] :runner: Improve notebook “scrolling up” rendering performance [vscode-148164](https://github-com/microsoft/vscode/-/148164) -roblourens
- [ ] :muscle: Explore improving files search results for notebooks [vscode-95275](https://github-com/microsoft/vscode/-/95275) -roblourens
- [ ] Investigate support for hot exit in interactive window [vscode-151444](https://github-com/microsoft/vscode/-/151444) -lramos15 -rebornix
- [ ] :red_circle: Improve selecting a renderer when multiple are registered for the same output mime-type -rebornix

---- Jupyter Notebooks
- [ ] :red_circle: Explore improve first-run experience for running a Jupyter notebook -brettcannon -lramos15 -IanMatthewHuff -karrtikr -minsa110
- [ ] :red_circle: :runner: Contributions to the vscode-jupyter extension -greazer **team**

--- Languages

---- AnyCode
- None.

---- LSP
- None.

---- Python
- [ ] :red_circle: Contributions to the python extension -brettcannon **team**
- [ ] :red_circle: Resolve language server restart - with `pylance` -dbaeumer

---- JavaScript/TypeScript
- [ ] :hand: Support TS server plugins on web [vscode-140455](https://github-com/microsoft/vscode/-/140455) -mjbvz
- [ ] :hand: Support TS project-wide intellisense on web [typescript-47600](https://github-com/microsoft/TypeScript/-/47600) -mjbvz
- [ ] :hand: Explore automatic type acqusition on web [typescript-45314](https://github-com/microsoft/TypeScript/-/45314) -mjbvz

---- Markdown
- [ ] :runner: Finalize DropIntoEditor API for creating file and image links [vscode-142990](https://github-com/microsoft/vscode/-/142990) -mjbvz
- [ ] :runner: Finalize DocumentPasteEditProvider API to support pasting images [vscode-30066](https://github-com/microsoft/vscode/-/30066) -mjbvz

---- CSS
- None.

---- JSON
- [ ] :muscle: Explore test suite from JSON schema spec [vscode-json-languageservice-120](https://github-com/microsoft/vscode-json-languageservice/-/102) -aeschli

--- Debug
- [x] :runner: Show breakpoint diagnostic information in the breakpoints view [vscode-142870](https://github-com/microsoft/vscode/-/142870) -roblourens
- [ ] Support to pass unescaped arguments in launch.json [vscode-149910](https://github-com/microsoft/vscode/-/149910) -roblourens -weinand
- [ ] Improving the UI for "step in targets" [vscode-149871](https://github-com/microsoft/vscode/-/149871) -roblourens 
- [ ] :runner: Remove special treatment of extension debugging in VS Code (Prototype extension debugging in a clean environment) [vscode-126737](https://github-com/microsoft/vscode/-/126737) -weinand

---- DAP
- [ ] Support to pass unescaped program arguments [debug-adapter-protocol-146](https://github-com/microsoft/debug-adapter-protocol/-/146) -weinand -roblourens
- [ ] Add additional location information to StepInTarget [debug-adapter-protocol-274](https://github-com/microsoft/debug-adapter-protocol/-/274) -connor4312 -weinand

---- js-debug
- [ ] :runner: :muscle: Investigate: Smart Step Into [vscode-123879](https://github-com/microsoft/vscode/-/123879) -connor4312
- [ ] Allow easy toggling between sourcemaps and compiled code [vscode-151412](https://github-com/microsoft/vscode/-/151412) -connor4312
- [ ] :muscle: Improve breakpoint predictor caching for large projects [vscode-js-debug-1149](https://github-com/microsoft/vscode-js-debug/-/1149) -connor4312
- [ ] :muscle: Automatically create debug config with a "dry run" [vscode-js-debug-1214](https://github-com/microsoft/vscode-js-debug/-/1214) -connor4312

--- Testing
- None.

--- Terminal
- [ ] :red_circle: Prepare to make shell integration the default -tylerleonhardt -meganrogge
- [ ] :red_circle: Address - with terminal reconnection -tyriar
- [ ] Improve accessibility in the terminal [vscode-149882](https://github-com/microsoft/vscode/-/149882-issuecomment-1134946927) -tyriar
- [ ] :red_circle: Explore contextual actions in the terminal -tyriar

--- Tasks
- [ ] :runner: Change feature area ownership of tasks [vscode-149060](https://github-com/microsoft/vscode/-/149060) -meganrogge -tyriar -alexr00
- [ ] Explore reconnecting tasks on window reload [vscode-117408](https://github-com/microsoft/vscode/-/117408) -meganrogge
- [ ] Support configurable icons and color for tasks [vscode-145965](https://github-com/microsoft/vscode/-/145965) -meganrogge

--- Source Control
- [x] :runner: Full editor can be used as git commit message editor [vscode-30562](https://github-com/microsoft/vscode/pull/30562) -lszomoru
- [ ] :muscle: Investigate: Support input box spell checking [vscode-35571](https://github-com/microsoft/vscode/-/35571) -lszomoru

--- API
- API finalization: [query](https://github-com/Microsoft/vscode/-?q=is%3Aissue+is%3Aopen+milestone%3A%22June+2022%22+label%3A%22api-finalization%22) -jrieken -mjbvz
- API proposals: [query](https://github-com/Microsoft/vscode/-?q=is%3Aissue+is%3Aopen+milestone%3A%22June+2022%22+label%3A%22api-proposal%22) -jrieken -mjbvz
- [ ] :red_circle: :muscle: Support snippets in workspace edits -jrieken
- [ ] :runner: Continue finalization of remaining proposed notebook API [vscode-149861)](https://github-com/microsoft/vscode/-/149861) -mjbvz -rebornix -roblourens
- [ ] Finalize API to disable the Source Control input field [vscode-150268](https://github-com/microsoft/vscode/-/150268) -lszomoru

--- Extensions
- [x] :runner: Provide a filter to list all deprecated extensions [vscode-151729](https://github-com/microsoft/vscode/-/151729) -sandy081

--- Extension Contributions
- [ ] GitHub Repositories: [query](https://github-com/microsoft/vscode-remote-repositories-github/milestone/15) -joyceerhl
- [ ] :red_circle: Improvements to the Remote extensions -tanhakabir -chrmarti -aeschli
- [ ] :runner: GHPR: Improvements papercuts, selfhosting -, [query](https://github-com/microsoft/vscode-pull-request-github/-?q=is%3Aopen+is%3Aissue+milestone%3A%22June+2022%22) -alexr00
- [ ] Close the loop between committing and creating a PR [vscode-151589](https://github-com/microsoft/vscode/-/151589) -alexr00 -lszomoru

--- Install / Update
- [ ] Windows: Unable to update [vscode-147408](https://github-com/microsoft/vscode/-/147408) -joaomoreno
- [ ] Windows: Prevent update service from activating if running as Admin in User install [vscode-148428](https://github-com/microsoft/vscode/-/148428) -joaomoreno
- [ ] :runner: Integrate with the Windows 11 explorer context menu [vscode-127365](https://github-com/microsoft/vscode/-/127365) -deepak1556

--- Engineering
- [ ] Sandbox: enable extension host communication [vscode-135268](https://github-com/microsoft/vscode/-/135268) -deepak1556 -alexdima
- [ ] :runner: Review errors to identify which ones are actionable as error telemetry [vscode-145043](https://github-com/microsoft/vscode/-/145043) -lramos15 **team**
- [ ] :red_circle: Explore spliting compilation and transpilation into two distinct gulp tasks -jrieken
- [ ] Explore using `swc` for transpilation [vscode-150025](https://github-com/microsoft/vscode/-/150025) -jrieken
- [x] :runner: Execute tests in parallel in CI build [vscode-151740](https://github-com/microsoft/vscode/-/151740) -lszomoru
- [ ] :muscle: Enable `contextIsolation` for all windows [vscode-128099](https://github-com/microsoft/vscode/-/128099) -bpasero -rzhao271 -deepak1556

---- Electron
- [x] :runner: Electron 18 update [vscode-145527](https://github-com/microsoft/vscode/pull/145527) -deepak1556
- [ ] :runner: Improve dependency generation for Debian [vscode-13089](https://github-com/microsoft/vscode/-/13089) -rzhao271
- [ ] WCO on Windows don't respond to touch events [electron-34364](https://github-com/electron/electron/-/34364) -deepak1556
- [ ] WCO on Windows occludes dev tools [electron-34129](https://github-com/electron/electron/-/34129) -deepak1556

--- Documentation
- [ ] :runner: Improve TS docs [vscode-docs-5205](https://github-com/microsoft/vscode-docs/-/5205) -gregvanl -isidorn

----
--- Deferred
- [ ] Allow resizing for the find widget in notebooks [vscode-141516](https://github-com/microsoft/vscode/-/141516) -rebornix
- [ ] Support copying simple output to the clipboard [vscode-129880](https://github-com/microsoft/vscode/-/129880) -amunger -rebornix
- [ ] Show changes in the overview ruler of the notebook diff editor [vscode-146224](https://github-com/microsoft/vscode/-/146224) -amunger -rebornix
- [ ] :runner: Provide support for Go to Test/Go to Test Subject [vscode-126932](https://github-com/microsoft/vscode/-/126932) -connor4312
- [ ] Indicate focused tab when using multiple editor groups in HC themes [vscode-145563](https://github-com/microsoft/vscode/-/145563) -daviddossett
- [ ] Align editor find widget with the notebook find widget [vscode-144851](https://github-com/microsoft/vscode/-/144851) -rebornix
- [ ] Make the eslint extension more approachable for new contributors [vscode-eslint-1447](https://github-com/microsoft/vscode-eslint/-/1447) -dbaeumer
- [ ] Investigate eslint notebook support in the web [vscode-eslint-1366](https://github-com/microsoft/vscode-eslint/-/1366) -dbaeumer
- [ ] Draft Git Beginner's Docs [vscode-docs-5033](https://github-com/microsoft/vscode-docs/-/5033) -digitarald -burkeholland
